qa/rpc-tests/bigblocks.py

```diff
@@ -151,12 +155,17 @@ def copy_mempool(self, from_node, to_node):
             txdata = from_node.getrawtransaction(txid)
             to_node.sendrawtransaction(txdata)
 
-    def TestMineBig(self, expect_big):
+    def TestMineBig(self, expect_big, expect_version=None):
         # Test if node0 will mine big blocks.
         b1hash = self.nodes[0].generate(1)[0]
         b1 = self.nodes[0].getblock(b1hash, True)
         assert(self.sync_blocks(self.nodes))
 
+        if expect_version:
+            assert b1['version'] & FORK_BLOCK_BIT
+        elif not expect_version==None:
+            assert not b1['version'] & FORK_BLOCK_BIT
+
         if expect_big:
             assert(b1['size'] > 1000*1000)
```

Comment:
If you have 4 MB of transactions in your mempool (as is the case in bigblocks.py), and if your node is configured to make blocks up to 2 MB in size, and it does not generate a > 1 MB block when it is allowed to, then there is a bug and the test should fail.

Note that bigblocks.py is part of the unit test code. Its purpose is to help us find bugs. It should fail when anything happens other than what we want or expect to happen.